runtime/java/src/common/Terminalton.java

```diff
@@ -0,0 +1,7 @@
+package common;
+
+
+public abstract class Terminalton<T> {
```

Comment:
Now there's two of them!  This is getting out of hand :upside_down_face: 

Seems reasonable to handle terminals in this way as well, but definitely needs a doc comment explaining why (and the name...)